internal/rbd/controllerserver.go

```diff
@@ -507,6 +507,12 @@ func (cs *ControllerServer) createBackingImage(ctx context.Context, cr *util.Cre
 		}
 	}
 	if rbdVol.Encrypted {
+		err = util.StoreNewCryptoPassphrase(rbdVol.VolID, rbdVol.KMS)
```

Comment:
The same check need to be added if the volume already exists scenario.

Reply:
Do you mean for static created volumes? In what case do volumes already exist?

Reply:
the volume already created during CreateVolume but the cephcsi plugin restarted when the next request comes in now both omap and rbd image exists but the KMS key is not generated yet?